### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by Hugoms-2001**

### CODE REVIEW

File: Assignment4/Hugoms-2001/task06.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+for s, p, o in g:
+  print(s,p,o)
+
+
+# In[9]:
+
+
+#TASK 6.4: Add to the individual JaneSmith the email address, fullName, given and family names
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+g.add((ns.JaneSmith, VCARD.hasEmail, Literal("jsmith@example.org")))
```
Comment (dgarijo) [inline Assignment4/Hugoms-2001/task06.py:79]: VCARD does not have this property. Look in the VCARD vocabulary for the right one to use.

File: Assignment4/Hugoms-2001/task07.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+
+#TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+
+print("SPARQL:")
+q1 = prepareQuery('''
+  SELECT ?SubClass WHERE { 
+    ?SubClass rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/Hugoms-2001/task07.py:39]: this does not address all subclasses of subclasses of living thing. Use `*`

File: Assignment4/Hugoms-2001/task07.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  }
+  ''',
+  initNs = { "rdfs": RDFS, "ns": ns}
+)
+for r in g.query(q1):
+  print(r)
+
+print("RDFLib:")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
```
Comment (dgarijo) [inline Assignment4/Hugoms-2001/task07.py:48]: same problem as above. Hint: make it recursive

File: Assignment4/Hugoms-2001/task07.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+for r in g.query(q2):
+  print(r)
+
+print("RDFLib:")
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
+    print(s)
+
+for s,p,o in g.triples ((None, RDFS.subClassOf, ns.Person)):
+    for s0,p0,o0 in g.triples((None, RDF.type, s)):
+        print (s0)
```
Comment (dgarijo) [inline Assignment4/Hugoms-2001/task07.py:75]: Same issue as above

File: Assignment4/Hugoms-2001/task07.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+#TASK 7.4: List the name of the persons who know Rocky
+
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+q4 = prepareQuery('''
+    SELECT ?Subject ?p ?o
+    WHERE {
+    ?Subject rdf:type ns:Person.
+    ?Subject foaf:knows ns:RockySmith.
```
Comment (dgarijo) [inline Assignment4/Hugoms-2001/task07.py:124]: the exercise is asking to return the name of the persons. You are not doing that

File: Assignment4/Hugoms-2001/task07.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+)
+
+unique_entities = set()
+
+for result in g.query(q5):
+    unique_entities.add(result.entity)
+
+for entity in unique_entities:
+    print(entity)
+
```
Comment (dgarijo) [inline Assignment4/Hugoms-2001/task07.py:156]: Missing 7.4 and 7.5 in rdflib. Please add them

**Review by dgarijo — CHANGES_REQUESTED** [review]
Changes required. Please have a look

**Comment by dgarijo** [comment]
latest commit did not solve issues

**Comment by anaigmo** [comment]
Submitted and fixed, no more PRs allowed

**Pull request merged**